src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -0,0 +1,200 @@
+
+    constructor() {
+        super();
+        super.id = 'vizshonenjump';
+        super.label = 'Viz - Shonen Jump';
+        this.tags = [ 'manga', 'english' ];
+        this.url = 'https://www.viz.com/shonenjump';
+        this.requestOptions.headers.set( 'x-referer', this.url );
+        this.config = {
+            /* Since the website requires us to fetch the image data a few seconds after the API call
+            *  we need to ensure we can download the data fast enough.
+            *  Also the website itself only loads a maximum of 4 pages at a time to ensure that.
+            */
+            concurrent: {
+                label: 'Concurrent downloads',
+                description: 'The maximum number of concurrent downloads.',
+                input: 'numeric',
+                min: 1,
+                max: 12,
+                value: 5
+            }
+        };
+        this.concurrent = 0;
+    }
+
+    async _getMangas() {
+        const request = new Request(new URL(this.url), this.requestOptions);
+        const data = await this.fetchDOM(request, 'div.o_sort_container div > a');
+
+        return data.map(manga => {
+            return {
+                id: manga.pathname,
+                title: manga.querySelector(':nth-child(2)').innerText.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        const request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, 'a.o_chapter-container');
+        data = data.filter(chapter => chapter.innerText.includes('FREE'));
+
+        return data.map(chapter => {
+            try {
+                return {
+                    id: chapter.href,
+                    title: chapter.querySelector('.disp-id').innerText.trim()
+                };
+            } catch(error) {
+                try {
+                    return {
+                        id: chapter.href,
+                        title: 'Ch. ' + chapter.href.match(/chapter-(\d+)\//)[1]
+                    };
+                } catch(error) {
+                    throw new Error('Unknown chapter format. Please report at https://github.com/manga-download/hakuneko/issues');
+                }
+            }
+        });
+    }
+
+    async _getPages(chapter) {
+        let script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    try {
+                        window.loadPages = function loadPages() {
+                            var aj_images = [];
+                            for (var page_count = 0; page_count <= pages; page_count++) 
+                                if (!(page_count < 0 || page_count > pages) && "undefined" == typeof pageImages["page" + page_count]) {
+                                    var aj_req = {
+                                        url: pageUrl + "&manga_id=" + manga_id + "&page=" + page_count,
```

Comment:
Is my assumption correct, that these links are only valid for a certain amount of time?
In that case i would recommend a different solution and just get the URL when the image is request in `_handleConnectorURI`

Reply:
At this point in the communication they are not yet time sensitive. The time sensitive API request is already inside of `_handleConnectorURI()`.